src/main/java/com/google/devtools/build/lib/packages/RuleClass.java

```diff
@@ -762,6 +762,10 @@ public Builder(String name, RuleClassType type, boolean skylark, RuleClass... pa
 
         for (Attribute attribute : parent.getAttributes()) {
           String attrName = attribute.getName();
+          if (attrName.equals(RuleClass.EXEC_COMPATIBLE_WITH_ATTR)) {
+            // Don't inherit: this will be re-created
```

Comment:
What's the remaining case for ExecutionPlatformConstraintsAllowed.PER_RULE?

i.e. can we just universally set this attribute in BaseRuleClasses now and let everything inherit it? Does the Starlark / native distinction get in the way?

Reply:
It's going to go away, and will only remain to allow starlark rules to turn it off until the `exec_compatible_with_allowed` attribute is removed (see #8136).

Reply:
Could we add a TODO to eventually then just set this attribute directly in the base rule class so we don't need this special exemption for it?

Reply:
The reason this appears here is because of ToolchainRule, which defines its own "exec_compatible_with", which is the same in practice but is used (and documented) differently.

Eventually just declaring the attribute, and letting ToolchainRule override it, is the right thing to do, so adding that TODO.

Reply:
Makes sense, thanks.